### PR TITLE
#11333: Resolve hang with Trace and R-Chip Event Synchronization

### PR DESCRIPTION
**Pull request by tt-asaigal**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/11333

### Problem description
  - Issuing a prefetch_h wait_for_event followed by a Trace command was causing data corruption on the remote chip
  - This was due to the Prefetcher coalescing the Trace cmd stream with the wait command. The wait command (along with its packet header) was being locally processed by prefetch_h, while the Trace cmd was sent to Mux without a header -> cmd stream was corrupted by Mux

### What's changed
  - The wait command is now locally processed by prefetch_h and sent to Mux, since it contains the header when commands are coalesced. Prefetch_d will process the wait as no-op
  - This can be made simpler once we have VCs on the packet network
### Checklist
- [x] Post commit CI passes
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: tt_metal/impl/dispatch/kernels/cq_prefetch.cpp [hunk 1]
```diff
@@ -1160,7 +1160,9 @@ bool process_cmd(uint32_t& cmd_ptr,
         }
         stride = process_debug_cmd(cmd_ptr);
         break;
-
+    case CQ_PREFETCH_CMD_WAIT_FOR_EVENT:
+        stride = CQ_PREFETCH_CMD_BARE_MIN_SIZE;
+        break;
```
Comment (pgkeller) [inline tt_metal/impl/dispatch/kernels/cq_prefetch.cpp:1165]: pedantic: missing new line below here to match the other cases

**Review by pgkeller — APPROVED** [review]

File: tt_metal/impl/dispatch/kernels/cq_prefetch.cpp (hunk 1 above)
Comment (tt-asaigal) [reply tt_metal/impl/dispatch/kernels/cq_prefetch.cpp:1166]: Updated, thanks!

**Review by tt-aho — APPROVED** [review]

File: tests/scripts/t3000/run_t3000_unit_tests.sh [hunk 2]
```diff
@@ -34,7 +34,6 @@ run_t3000_ttnn_tests() {
   echo "LOG_METAL: Running run_t3000_ttnn_tests"
   WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml ./build/test/ttnn/test_multi_device
   WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml ./build/test/ttnn/unit_tests_ttnn
-  pytest -n auto tests/ttnn/unit_tests/test_multi_device_trace.py ; fail+=$?
```
Comment (ttmchiou) [inline tests/scripts/t3000/run_t3000_unit_tests.sh:37]: Is this suppose to be removed?

**Review by TT-BrianLiu — APPROVED** [review]

File: tests/scripts/t3000/run_t3000_unit_tests.sh (hunk 2 above)
Comment (tt-asaigal) [reply tests/scripts/t3000/run_t3000_unit_tests.sh:37]: Yes, I'm doubling the number of tests instantiated by calling:
`WH_ARCH_YAML=wormhole_b0_80_arch_eth_dispatch.yaml pytest -n auto tests/ttnn/unit_tests/test_multi_device_trace.py ; fail+=$?`, so it made sense to remove some tests to keep CI runtime reasonable. The tests being removed are slightly redundant at this point.

**Review by ttmchiou — APPROVED** [review]

**Pull request merged**